lib/fuzzy-finder-view.js

```diff
@@ -118,16 +149,38 @@ class FuzzyFinderView {
 
     this.subscriptions = new CompositeDisposable()
     this.subscriptions.add(
-      atom.config.observe('fuzzy-finder.useAlternateScoring', (newValue) => {
-        this.useAlternateScoring = newValue
-        if (this.useAlternateScoring) {
+      atom.config.observe('fuzzy-finder.scoringSystem', (newValue) => {
+        this.scoringSystem = newValue
+
+        if (this.scoringSystem === SCORING_SYSTEMS.STANDARD) {
+          this.selectListView.update({filter: null})
+        } else if (this.scoringSystem === SCORING_SYSTEMS.FAST) {
+          if (!this.nativeFuzzy) {
+            this.nativeFuzzy = new NativeFuzzy.Matcher(this.items.map(el => el.label))
+
+            // We need a separate instance of the fuzzy finder to calculate the
+            // matched paths only for the returned results. This speeds up considerably
+            // the filtering of items.
+            this.nativeFuzzyForResults = new NativeFuzzy.Matcher([])
+          }
+
           this.selectListView.update({
             filter: (items, query) => {
-              return query ? fuzzaldrinPlus.filter(items, query, {key: 'label'}) : items
+              if (!query) {
+                return items
+              }
+
+              return this.nativeFuzzy.match(query, {maxResults: MAX_RESULTS}).map(
```

Comment:
For now, we're executing `native-fuzzy` on a single thread, but we can easily change this in the future